analysis/study_definition.py

```diff
@@ -1,13 +1,884 @@
+      },
+  )
+def imae_snomed(dx_codelist):  ## to idenitfy NEW conditions, will include all first diagnosis but then on stata remove those diagnosed before covid test positive date
+  return patients.with_these_clinical_events(
+      dx_codelist,
+      returning="date",
+      on_or_before ="covid_test_positive_date + 28 days",
+      find_first_match_in_period=True,
+      date_format="YYYY-MM-DD",
+      return_expectations={
+          "incidence": 0.2,
+          "date": {"earliest": "2021-12-16"},
+      },
+  )
+def imae_icd(dx_codelist):  ## to idenitfy NEW conditions, will include all first diagnosis but then on stata remove those diagnosed before covid test positive date
+  return patients.admitted_to_hospital(
+      with_these_diagnoses = dx_codelist,
+      returning="date_admitted",
+      on_or_before ="covid_test_positive_date + 28 days",
+      find_first_match_in_period=True,
+      date_format="YYYY-MM-DD",
+      return_expectations={
+          "incidence": 0.2,
+          "date": {"earliest": "2021-12-16"},
+      },
+  )
 study = StudyDefinition(
-    default_expectations={
-        "date": {"earliest": "1900-01-01", "latest": "today"},
-        "rate": "uniform",
-        "incidence": 0.5,
+  # Configure the expectations framework
+  default_expectations = {
+    "date": {"earliest": "2021-11-01", "latest": "today"},
+    "rate": "uniform",
+    "incidence": 0.05,
+  },
+      
+  ## Define index date
+  index_date = campaign_start,
+
+  population = patients.satisfying(
+    """
+    age >= 18 AND age < 110
+    AND NOT has_died
+    AND covid_test_positive
+    AND eligible_binary
+    """,
+  ),
+        
+  ## COVID TEST POSITIVE
+  ## First positive SARS-CoV-2 test. [patients are eligible for treatment if diagnosed <=5d ago. Restricted to first positive covid test after index date]
+  covid_test_positive = patients.with_test_result_in_sgss(
+    pathogen = "SARS-CoV-2", test_result = "positive", returning = "binary_flag", on_or_after = "index_date - 5 days", find_first_match_in_period = True, 
+    restrict_to_earliest_specimen_date = True,  return_expectations = {"incidence": 0.9 },
+  ),
+  covid_test_positive_date = patients.with_test_result_in_sgss(
+    pathogen = "SARS-CoV-2", test_result = "positive", returning = "date", date_format = "YYYY-MM-DD", on_or_after = "index_date - 5 days", find_first_match_in_period = True, 
+    restrict_to_earliest_specimen_date = True, return_expectations = {"date": {"earliest": "2021-12-11", "latest": "today"}, "incidence": 0.9},
+  ),
+  covid_positive_test_type = patients.with_test_result_in_sgss(
+    pathogen = "SARS-CoV-2", test_result = "positive", returning = "case_category", on_or_after = "index_date - 5 days", find_first_match_in_period = True, 
+    restrict_to_earliest_specimen_date = True, return_expectations = {"category": {"ratios": {"LFT_Only": 0.4, "PCR_Only": 0.4, "LFT_WithPCR": 0.2}},"incidence": 0.2, },
+  ),
+  ## Second positive SARS-CoV-2 test
```

Comment:
Might be worth doing some exploratory analysis to see how many times people are testing positive (as potentially excluding individuals if limiting to 1/2 covid tests)

Reply:
Yes this is a good point, I dont think they did this for the efficacy analysis
This would certainly increase the treatment arms, but need to make sure that these patient don't act as a control on their first positive test and then a treatment arm in their second test. 

I could check this first